lib/persistence/mongo.js

```diff
@@ -187,22 +187,25 @@ MongoPersistence.prototype.storeSubscriptions = function(client, done) {
       return client.subscriptions[key].qos > 0;
     });
 
-    steed.each(subscriptions, function(key, cb) {
-      that._subscriptions.findAndModify({
-        client: client.id,
-        topic: key
-      }, [['date', -1]], {
-        $set: {
+    this._subscriptions.deleteMany({client: client.id}, function(err, subs){
+      if (err) return done(err);
+      var save_subs = [];
+      subscriptions.forEach(function(key){
+        save_subs.push({
           client: client.id,
           topic: key,
           qos: client.subscriptions[key].qos,
           added: new Date()
-        }
-      }, { upsert: true}, cb);
-    }, done);
-  } else if (done) {
-    return done();
+        });
+      });
+
+      if (save_subs.length > 0) {
+        that._subscriptions.insertMany(save_subs, {w: 1}, done);
+        return;
+      }
+    });
   }
+  done && done();
```

Comment:
I would prefer a different fix that did not require to remove all the subscriptions of a client and adding all of them back, as the Node process could crash in between.